src/ln/channelmanager.rs

```diff
@@ -2661,6 +2665,20 @@ impl ChainListener for ChannelManager {
 		for failure in failed_channels.drain(..) {
 			self.finish_force_close_channel(failure);
 		}
+		{
+			let mut channel_state = Some(self.channel_state.lock().unwrap());
+			for htlc_update in self.monitor.fetch_pending_htlc_updated() {
+				if let Some(source) = htlc_update.source {
+					if let Some(preimage) = htlc_update.payment_preimage {
+						if channel_state.is_none() { channel_state = Some(self.channel_state.lock().unwrap());}
+						self.claim_funds_internal(channel_state.take().unwrap(), source, preimage);
+					} else {
+						if channel_state.is_none() { channel_state = Some(self.channel_state.lock().unwrap());}
+						self.fail_htlc_backwards_internal(channel_state.take().unwrap(), source, &htlc_update.payment_hash, HTLCFailReason::Reason { failure_code: 0x2000 | 2, data: Vec::new() });
```

Comment:
I think this can just fail with unknown_next_peer - that channel is closed so we should just tell them to never try to route over it again.

Reply:
Okay, corrected with PERM|10, but in fact it's only one of the channel which has been failed not the whole node (but if remote node has unilateral closed channel it's an indication that it's unreliable)